slsDetectorSoftware/src/DetectorImpl.cpp

```diff
@@ -1286,15 +1305,73 @@ void DetectorImpl::startAcquisition(bool blocking, std::vector<int> positions) {
                 Parallel(&Module::startAcquisition, master);
             }
         }
-    } else {
+    }
+    // all in parallel
+    else {
         if (blocking) {
-            Parallel(&Module::startAndReadAll, positions);
+            Parallel(&Module::startAndReadAll, pos);
         } else {
-            Parallel(&Module::startAcquisition, positions);
+            Parallel(&Module::startAcquisition, pos);
         }
     }
 }
 
+void DetectorImpl::sendSoftwareTrigger(const bool block, Positions pos) {
+    bool handleSynchronization = false;
+    // multi module sync enabled jungfrau
+    if (size() > 1 && (shm()->detType == defs::JUNGFRAU) &&
+        (Parallel(&Module::getSynchronizationFromStopServer, pos)
+             .tsquash("Inconsistent synchronization among modules"))) {
+        handleSynchronization = true;
+    }
+
+    // slaves first
+    if (handleSynchronization) {
```

Comment:
How does Junfrau/Mythen3 work? I think we only send software trigger to master (or all if non sync)

Reply:
From yesterdays meeting, JF must be slaves first, then master. For M3: it doesnt matter, so I will do all of them like JF. G2/G1 dont have it implemented, but that as you said can give an exception if used from the detetor server itself.

Reply:
Refactored software trigger for m3 as well.